File: examples/elixir/mega_flow/executor.ex
defmodule Mega.Executor do
  # SINK — :os.cmd · elixir.cmdi.os_cmd · CWE-78
  def execute(cmd) do
    :os.cmd(String.to_charlist(cmd))
    cmd
  end

  def clean_twin do
    # NEGATIVE — same sink kind with a constant argument must not report.
    :os.cmd(~c"echo clean")
    "clean"
  end
end

File: examples/erlang/mega_flow/executor.erl
-module(executor).
-export([execute/1, clean_twin/0]).

execute(Cmd) ->
    %% SINK — os:cmd · erlang.cmdi.os_cmd · CWE-78
    os:cmd(Cmd),
    Cmd.

clean_twin() ->
    %% NEGATIVE — same sink kind with a constant argument must not report.
    os:cmd("echo clean"),
    "clean".
